e2e/vmss.go

```diff
@@ -203,16 +192,20 @@ func getNewRSAKeyPair() (privatePEMBytes []byte, publicKeyBytes []byte, e error)
 	return
 }
 
-func getVmssName() string {
-	return fmt.Sprintf(vmssNameTemplate, randomLowercaseString(4))
+func getVmssName(t *testing.T) string {
+	name := fmt.Sprintf("%s-%s-%s-%s", vmssNamePrefix, time.Now().Format(time.DateOnly), randomLowercaseString(4), t.Name())
+	// delete invalid characters like _ and /
+	name = strings.ReplaceAll(name, "_", "")
+	name = strings.ReplaceAll(name, "/", "")
+	name = strings.ReplaceAll(name, "Test", "")
+	// truncate to 58 characters
+	if len(name) > 58 {
```

Comment:
this would probably truncate the TestName, do we consider this portion of the name to be less important to help us identify ownership/source of the VMSS ? (in case the t.Name() gets truncated)

Reply:
🤷 I think it's less important. But it just must be unique between test runs.